### PR TITLE
remote-config: rename FEATURES test to ASM_FEATURES

### PR DESCRIPTION
**Pull request by ameske**
The product FEATURES was renamed to ASM_FEATURES prior to going prod, so this just updates all references of the old name so that new tracers can test against the product that will go live.

Closes: #558
Closes: #559 

## Description

Please include a short summary of the change

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

**Comment by cbeauchesne** [comment]
If i'm not wrong, these scnearios are in the CI, is it expected ?

File: run.sh [hunk 1]
```diff
@@ -144,10 +145,11 @@ elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_DD" ]; then
     export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_asm_dd
     export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "ASM_DD"}'
 
-elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_FEATURES_NOCACHE" ]; then
+elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_FEATURES_NOCACHE" ]; then
     export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationFields scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeaturesNoCache"
     export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_features_nocache
```
Comment (simon-id) [inline run.sh:150]:
```suggestion
    export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_asm_features_nocache
```

File: run.sh [hunk 2]
```diff
@@ -144,10 +145,11 @@ elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_DD" ]; then
...
-    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "FEATURES_NO_CACHE"}'
+    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "ASM_FEATURES_NO_CACHE"}'
+    WEBLOG_ENV="DD_APPSEC_ENABLED=false"
```
Comment (simon-id) [inline run.sh:152]: Same here, see comment above

**Comment by simon-id** [comment]
It looks like the file `scenarios/remote_config/rc_expected_requests.json` didn't get updated by the scenario generator script ? it still has the old `FEATURES` ?

https://github.com/DataDog/system-tests/blob/b562c9c351a9b46f4dade04c5ed9dabd6bc14c17/scenarios/remote_config/rc_expected_requests.json#L79

File: run.sh [hunk 3]
```diff
@@ -128,10 +128,11 @@ elif [ $SYSTEMTESTS_SCENARIO = "BACKEND_WAF" ]; then
     export SYSTEMTESTS_LOG_FOLDER=logs_backend_waf
     WEBLOG_ENV="DD_APPSEC_ENABLED=false"
 
-elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_FEATURES" ]; then
+elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_FEATURES" ]; then
     export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationFields scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeatures"
     export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_features
-    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "FEATURES"}'
+    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "ASM_FEATURES"}'
+    WEBLOG_ENV="DD_APPSEC_ENABLED=false"
```
[the remark this answers is not in the document]
Comment (simon-id) [reply run.sh:135]: I tried locally and it looks like `DD_APPSEC_ENABLED=` works:
```suggestion
    WEBLOG_ENV="DD_APPSEC_ENABLED="
```

File: run.sh (hunk 2 above)
Comment (simon-id) [reply run.sh:152]:
```suggestion
    WEBLOG_ENV="DD_APPSEC_ENABLED="
```

**Review by simon-id — CHANGES_REQUESTED** [review]

File: run.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (ameske) [reply run.sh:135]: Ohhhh, whoops...yeah that makes more sense now reading the issue again. Thanks!

**Comment by ameske** [comment]
> It looks like the file scenarios/remote_config/rc_expected_requests.json didn't get updated by the scenario generator script ? it still has the old FEATURES ? I don't know if this file is used tbh ?

It looks like it is not used, an artifact from when we didn't have multiple test scenarios. I'll make another PR to remove it.

**Review by simon-id — APPROVED** [review]

**Comment by ameske** [comment]
> If i'm not wrong, these scnearios are in the CI, is it expected ?

I think everything is fine since we isolated the remote config tests to independent executions, if I'm understanding "scenarios" correctly.

**Comment by cbeauchesne** [comment]
> > If i'm not wrong, these scnearios are in the CI, is it expected ?
> 
> @cbeauchesne - We don't want disabling APPSEC to cause issues with other tests but I think everything is fine since we isolated the remote config tests to independent executions, if I'm understanding "scenarios" correctly.

Scenario runs are totally isolated (it's their purpose :) )

My point is that those two scenario are never executed in the system tests ci, meaning that we can break them anyday without knowing it. Can you add them in `.github/workflow/ci.yml` ?

**Comment by simon-id** [comment]
@cbeauchesne 
>If i'm not wrong, these scnearios are in the CI, is it expected ?

I think you forgot a "not" :smile:

**Comment by cbeauchesne** [comment]
> I think you forgot a "not" 😄

indeed

**Comment by ameske** [comment]
Can definitely do this, but might require some more discussion. We have a single "scenario" but multiple flavors of it. (Basically one for each product, and then for each product the "cached" and "no cached" version) We can't run multiple flavors together because of the stateful nature of remote config.

Thus to test everything we'd need this block for all possible "flavors":

```yml
 - name: Run REMOTE_CONFIG_MOCKED_BACKEND_<PRODUCT>_<FLAVOR> scenario
   run: ./run.sh REMOTE_CONFIG_MOCKED_BACKEND_<PRODUCT>_<FLAVOR>
   env:
       DD_API_KEY: ${{ secrets.DD_API_KEY }}
```

Does that sound right/ok to you?

**Comment by Julio-Guerra** [comment]
Hello,
Any chance we can move forward on this so that the ASM libs can rely on those tests? 🙏

**Comment by cbeauchesne** [comment]
@ameske how may product/flavor do you have ?

**Comment by ameske** [comment]
I just cleaned up the merge conflict, assuming CI/CD passes here let's go ahead and merge this so that discussion around adding the scenarios to the CI doesn't slow down progress. I'll open a PR with the naive route of registering scenarios with CI/CD and we can take a look and see if there's a better way.

**Comment by ameske** [comment]
Trying to help move this along, but I'm not sure what's going on with all the failures at the moment. I'm trying to reproduce locally but not having any success - the runner is just hanging.

**Comment by cbeauchesne** [comment]
I'll have a look

File: run.sh [hunk 4]
```diff
@@ -122,10 +122,17 @@ elif [ $SYSTEMTESTS_SCENARIO = "LIBRARY_CONF_CUSTOM_HEADERS_LONG" ]; then
     DD_TRACE_HEADER_TAGS=$(docker run system_tests/weblog env | grep DD_TRACE_HEADER_TAGS | cut -d'=' -f2)
     WEBLOG_ENV="DD_TRACE_HEADER_TAGS=$DD_TRACE_HEADER_TAGS,header-tag1:custom.header-tag1,header-tag2:custom.header-tag2"
 
-elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_FEATURES" ]; then
-    export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeatures"
-    export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_features
-    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "FEATURES"}'
+elif [ $SYSTEMTESTS_SCENARIO = "BACKEND_WAF" ]; then
```
Comment (simon-id) [inline run.sh:125]: @cbeauchesne Is this a merge conflict artifact ? should we remove it ?

File: run.sh [hunk 5]
```diff
@@ -138,10 +145,11 @@ elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_DD" ]; then
     export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_asm_dd
     export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "ASM_DD"}'
 
-elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_FEATURES_NOCACHE" ]; then
-    export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeaturesNoCache"
-    export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_features_nocache
-    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "FEATURES_NO_CACHE"}'
+elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_FEATURES_NOCACHE" ]; then
+    export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationFields scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeaturesNoCache"
```
Comment (simon-id) [inline run.sh:149]:
```suggestion
    export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeaturesNoCache"
```

File: run.sh [hunk 6]
```diff
@@ -122,10 +122,17 @@ elif [ $SYSTEMTESTS_SCENARIO = "LIBRARY_CONF_CUSTOM_HEADERS_LONG" ]; then
...
-    export SYSTEMTESTS_LOG_FOLDER=logs_remote_config_mocked_backend_features
-    export SYSTEMTESTS_LIBRARY_PROXY_STATE='{"mock_remote_config_backend": "FEATURES"}'
+elif [ $SYSTEMTESTS_SCENARIO = "BACKEND_WAF" ]; then
+    # disable appsec
+    export RUNNER_ARGS=scenarios/backend/test_waf.py
+    export SYSTEMTESTS_LOG_FOLDER=logs_backend_waf
+    WEBLOG_ENV="DD_APPSEC_ENABLED=false"
+
+elif [ $SYSTEMTESTS_SCENARIO = "REMOTE_CONFIG_MOCKED_BACKEND_ASM_FEATURES" ]; then
+    export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationFields scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeatures"
```
Comment (simon-id) [inline run.sh:132]: @ameske I found the issue:
```suggestion
    export RUNNER_ARGS="scenarios/remote_config/test_remote_configuration.py::Test_RemoteConfigurationUpdateSequenceFeatures"
```

File: run.sh (hunk 4 above)
Comment (smola) [reply run.sh:125]: AFAIK this PR did not add this originally, and it should be merge conflict issue.

Comment (simon-id) [reply run.sh:125]: I think so too, let's wait for the others to confirm

Comment (ameske) [reply run.sh:125]: I must have brought it in accidentally when I tried to get the branch up to date against main. Sorry about that!

**Review by simon-id — APPROVED** [review]

File: utils/proxy/forwarder.py [hunk 7]
```diff
@@ -173,16 +173,16 @@ def response(self, flow):
...
+            self._modify_response_rc(flow, RC_MOCKED_RESPONSES_ASM_FEATURES)
         elif self.state.get("mock_remote_config_backend") == "LIVE_DEBUGGING":
             self._modify_response_rc(flow, RC_MOCKED_RESPONSES_LIVE_DEBUGGING)
         elif self.state.get("mock_remote_config_backend") == "ASM_DD":
             self._modify_response_rc(flow, RC_MOCKED_RESPONSES_ASM_DD)
         elif self.state.get("mock_remote_config_backend") == "ASM_DATA":
             self._modify_response_rc(flow, RC_MOCKED_RESPONSES_ASM_DATA)
-        if self.state.get("mock_remote_config_backend") == "FEATURES_NO_CACHE":
-            self._modify_response_rc(flow, RC_MOCKED_RESPONSES_FEATURES_NO_CACHE)
+        if self.state.get("mock_remote_config_backend") == "ASM_FEATURES_NO_CACHE":
```
Comment (simon-id) [inline utils/proxy/forwarder.py:184]: missing else if no ?
```suggestion
        elif self.state.get("mock_remote_config_backend") == "ASM_FEATURES_NO_CACHE":
```

Comment (simon-id) [reply utils/proxy/forwarder.py:184]: @cbeauchesne

**Comment by ameske** [comment]
Seems the current failures all stem from an issue on main, and not this PR. @cbeauchesne - we're all set with this specific change so we're just needing a system-tests core review now.

File: utils/proxy/forwarder.py (hunk 7 above)
Comment (ameske) [reply utils/proxy/forwarder.py:184]: Ah, yes. No need to unconditionally always check this 😁 .

**Review by simon-id — APPROVED** [review]

**Comment by simon-id** [comment]
Status update: the file `./run.sh` was somehow chmod'ed to non executable, so I've changed it back

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**